adapters/algorix/algorix.go

```diff
@@ -0,0 +1,166 @@
+package algorix
+
+import (
+	"encoding/json"
+	"fmt"
+	"github.com/prebid/prebid-server/errortypes"
+	"github.com/prebid/prebid-server/macros"
+	"net/http"
+	"text/template"
+
+	"github.com/mxmCherry/openrtb/v15/openrtb2"
+	"github.com/prebid/prebid-server/adapters"
+	"github.com/prebid/prebid-server/config"
+	"github.com/prebid/prebid-server/openrtb_ext"
+)
+
+type adapter struct {
+	EndpointTemplate template.Template
+}
+
+// Builder builds a new instance of the Foo adapter for the given bidder with the given config.
+func Builder(bidderName openrtb_ext.BidderName, config config.Adapter) (adapters.Bidder, error) {
+	endpoint, err := template.New("endpointTemplate").Parse(config.Endpoint)
+	if err != nil {
+		return nil, fmt.Errorf("unable to parse endpoint url template: %v", err)
+	}
+	bidder := &adapter{
+		EndpointTemplate: *endpoint,
+	}
+	return bidder, nil
+}
+
+func (a *adapter) MakeRequests(request *openrtb2.BidRequest, requestInfo *adapters.ExtraRequestInfo) ([]*adapters.RequestData, []error) {
+	var adapterRequests []*adapters.RequestData
+	var errs []error
+
+	adapterRequest, err := a.makeRequest(request)
+	if err == nil {
+		adapterRequests = append(adapterRequests, adapterRequest)
+	} else {
+		errs = append(errs, err)
+	}
+	return adapterRequests, errs
+}
+
+func (a *adapter) makeRequest(request *openrtb2.BidRequest) (*adapters.RequestData, error) {
+	algorixExt := parseAlgoriXExt(request)
+
+	if algorixExt == nil {
+		return nil, &errortypes.BadInput{Message: "Invalid ExtImpAlgoriX value"}
+	}
+
+	endPoint, err := a.getEndPoint(algorixExt)
+	if err != nil {
+		return nil, err
+	}
+
+	preProcess(request)
+	reqBody, err := json.Marshal(request)
+	if err != nil {
+		return nil, err
+	}
+
+	headers := http.Header{}
+	headers.Add("Content-Type", "application/json;charset=utf-8")
+	headers.Add("Accept", "application/json")
+	headers.Add("x-openrtb-version", "2.5")
+
+	return &adapters.RequestData{
+		Method:  "POST",
+		Uri:     endPoint,
+		Body:    reqBody,
+		Headers: headers,
+	}, nil
+}
+
+//parseAlgoriXExt parse AlgoriX Ext
```

Comment:
Nitpick: The comment isn't particularly descriptive, and it doesn't need to be in my opinion since the name is straightforward. Consider removing the comment.